catalogs/catalog-jdbc-mysql/src/test/java/com/datastrato/gravitino/catalog/mysql/integration/test/CatalogMysqlIT.java

```diff
@@ -1351,6 +1355,11 @@ void testMySQLSchemaNameCaseSensitive() {
               .stream()
               .anyMatch(n -> n.equals(tableName)));
     }
+
+    for (String schema : schemas) {
```

Comment:
After I had added a IT which `CatalogMysqlDriverIT` extends `CatalogMysqlIT`, There is a strange appearance on `CatalogMysqlIT#testNameSpec, testSchemaComment, testMySQLSchemaNameCaseSensitive` test. All of them run failed caused by `database exists` exception, because create a new database for themself test purpose.

Strange thing is that why added `CatalogMysqlDriverIT` test class affect `CatalogMysqlIT`, I think they all will run on seperated mysql container for `CatalogMysqlIT#startup()` method:

```
  @BeforeAll
  public void startup() throws IOException, SQLException {

    if (!ITUtils.EMBEDDED_TEST_MODE.equals(testMode)) {
      String gravitinoHome = System.getenv("GRAVITINO_HOME");
      Path tmpPath = Paths.get(gravitinoHome, "/catalogs/jdbc-mysql/libs");
      JdbcDriverDownloader.downloadJdbcDriver(mysqlDriverDownloadUrl, tmpPath.toString());
    }

    TEST_DB_NAME = TestDatabaseName.MYSQL_CATALOG_MYSQL_IT;

    if (mysqlImageName.equals("mysql:5.7")) {
      containerSuite.startMySQLVersion5Container(TestDatabaseName.MYSQL_CATALOG_MYSQL_IT);
      MYSQL_CONTAINER = containerSuite.getMySQLVersion5Container();
    } else {
      containerSuite.startMySQLContainer(TEST_DB_NAME);
      MYSQL_CONTAINER = containerSuite.getMySQLContainer();
    }

    ...
  }
```

I tried some ways but didn't find out the indeed reason or useful solutions, so just add a clear schema behavior on three tests

Reply:
I checked `containerSuite.startMySQLContainer(TEST_DB_NAME);`. I have a misunderstood that it didn't start a new container always, so new added `CatalogMysqlDriverIT` and extends `CatalogMysqlIT` IT using the same mysql container.

`CatalogMysqlIT#testNameSpec, testSchemaComment, testMySQLSchemaNameCaseSensitive` test create schema and don't clean them cause `database exists` exception

Reply:
Dropping the database after the tests won't guarantee success if they are run in parallel.  You might be able to confirm that the newly added tests CatalogMysqlDriverIT and CatalogMysqlIT always run in serialization.

Reply:
> Dropping the database after the tests won't guarantee success if they are run in parallel. You might be able to confirm that the newly added tests CatalogMysqlDriverIT and CatalogMysqlIT always run in serialization.

not IT run parallel but ITs use the same mysql container.

```
@Tag("gravitino-docker-it")
@TestInstance(Lifecycle.PER_CLASS)
public class CatalogMysqlIT extends AbstractIT {
  private static final ContainerSuite containerSuite = ContainerSuite.getInstance();
  ...
  private MySQLContainer MYSQL_CONTAINER;
  ...

  @BeforeAll
  public void startup() throws IOException, SQLException {
  ...
    if (mysqlImageName.equals("mysql:5.7")) {
      containerSuite.startMySQLVersion5Container(TestDatabaseName.MYSQL_CATALOG_MYSQL_IT);
      MYSQL_CONTAINER = containerSuite.getMySQLVersion5Container();
    } else {
      containerSuite.startMySQLContainer(TEST_DB_NAME);
      MYSQL_CONTAINER = containerSuite.getMySQLContainer();
    }
  ...
  }
  ...
```
`CatalogMysqlIT` and new added `CatalogMysqlDriverIT` use the same mysql container, `CatalogMysqlVersion5IT` start a new mysql 5.7 container that's why `CatalogMysqlVersion5IT` extends `CatalogMysqlIT` similarly and not affect `database exists` exception
```
      containerSuite.startMySQLContainer(TEST_DB_NAME);
      MYSQL_CONTAINER = containerSuite.getMySQLContainer();
```

`CatalogMysqlIT#stop` or `CatalogMysqlIT#resetSchema` method can't clear every test created schemas, only clear the global schema
```
public class CatalogMysqlIT extends AbstractIT {

  public String metalakeName = GravitinoITUtils.genRandomName("mysql_it_metalake");
  public String catalogName = GravitinoITUtils.genRandomName("mysql_it_catalog");
  public String schemaName = GravitinoITUtils.genRandomName("mysql_it_schema");
  public String tableName = GravitinoITUtils.genRandomName("mysql_it_table");

  @AfterAll
  public void stop() {
    clearTableAndSchema();
    client.dropMetalake(NameIdentifier.of(metalakeName));
    mysqlService.close();
  }

  @AfterEach
  public void resetSchema() {
    clearTableAndSchema();
    createSchema();
  }

  private void clearTableAndSchema() {
    NameIdentifier[] nameIdentifiers =
        catalog.asTableCatalog().listTables(Namespace.of(metalakeName, catalogName, schemaName));
    for (NameIdentifier nameIdentifier : nameIdentifiers) {
      catalog.asTableCatalog().dropTable(nameIdentifier);
    }
    catalog.asSchemas().dropSchema(NameIdentifier.of(metalakeName, catalogName, schemaName), false);
  }

```

Reply:
I think we should try to clear schemas created by every test or start a seperate mysql container for every IT.

to clear created schemas when tests finish, add clear behavior on every test last or the `@AfterEach` method. 
But `@AfterEach` method should be aware of what schemas to be cleared, clear behavior added on every test last may be better

Reply:
> not IT run parallel but ITs use the same mysql container.

I know that they use a shared container, so if they run serialization, dropping the database after each test case should work. If they run parallel, I believe it won't solve the problem completely. 
```text
Thread1.          Thread2

create db1.     create db1
drop db1         drop db1  
```

Reply:
> I think we should try to clear schemas created by every test or start a seperate mysql container for every IT.
> 
> to clear created schemas when tests finish, add clear behavior on every test last or the `@AfterEach` method. But `@AfterEach` method should be aware of what schemas to be cleared, clear behavior added on every test last may be better

Yes, this should be reasonable in any case.

Reply:
> > not IT run parallel but ITs use the same mysql container.
> 
> I know that they use a shared container, so if they run serialization, dropping the database after each test case should work. If they run parallel, I believe it won't solve the problem completely.
> 
> ```
> Thread1.          Thread2
> 
> create db1.     create db1
> drop db1         drop db1  
> ```

I have understood what you means, now `CatalogMysqlIT` and `CatalogMysqlDriverIT` run parallel and there is probability that `database exists` exception appears, but serialization will not. 

I try to find out how to do this

Reply:
junit5 runs serialization default, junit5 support parallel runs and should not be enabled on gravitino.

so clear schemas after tests finish will be ok.

what about your suggest ? @yuqi1129

Reply:
> junit5 runs serialization default, junit5 support parallel runs and should not be enabled on gravitino.
> 
> so clear schemas after tests finish will be ok.
> 
> what about your suggest ? @yuqi1129

Then, the current changes will resolve the problem. I'm okay with it. Let's wait for the CI to succeed.